client/http-client/src/client.rs

```diff
@@ -267,7 +270,8 @@ impl ClientT for HttpClient {
 					Some(pos) => *pos,
 					None => return Err(Error::InvalidRequestId),
 				};
-				responses[pos] = rp.result
+				let result = serde_json::from_value(rp.result).map_err(Error::ParseError)?;
```

Comment:
Same here; a comment for the two-part decoding and use RawValue if possible to avoid allocating?

Reply:
yeah, it works ty for the pro-tip :)